pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go

```diff
@@ -27,12 +28,23 @@ type KubeDirectorAppSpec struct {
 	SchemaVersion       int                 `json:"configSchemaVersion"`
 	DefaultImageRepoTag *string             `json:"defaultImageRepoTag,omitempty"`
 	DefaultSetupPackage SetupPackage        `json:"defaultConfigPackage,omitempty"`
-	Services            []Service           `json:"services,omitempty"`
+	Services            []Service           `json:"services"`
 	NodeRoles           []NodeRole          `json:"roles"`
 	Config              NodeGroupConfig     `json:"config"`
-	DefaultPersistDirs  *[]string           `json:"defaultPersistDirs,omitempty"`
-	Capabilities        []corev1.Capability `json:"capabilities,omitempty"`
-	SystemdRequired     bool                `json:"systemdRequired,omitempty"`
+	DefaultPersistDirs  *[]string           `json:"defaultPersistDirs"`
+	Capabilities        []v1.Capability     `json:"capabilities"`
+	SystemdRequired     bool                `json:"systemdRequired"`
+	ConnectableTo       []ConnectableConfig `json:"connectable_to,omitempty"`
+}
+
+// ConnectableConfig describes type of objects that can be connected to
+// a cluster that uses the app.
+// XXX FIXME. Only categor
```

Comment:
If this isn't used anywhere yet, let's remove it.

Reply:
yes. As mentioned below. Will do this in a later PR